be/src/storage/tablet_schema.h

```diff
@@ -320,6 +410,11 @@ class TabletSchema {
     Status build_current_tablet_schema(int64_t index_id, int32_t version, const POlapTableIndexSchema& index,
                                        const std::shared_ptr<const TabletSchema>& ori_tablet_schema);
 
+    const std::vector<TabletIndex>& indexes() const { return _indexes; }
+    Status get_indexes_for_column(int32_t col_unique_id, std::unordered_map<IndexType, TabletIndex>* res) const;
+    Status get_indexes_for_column(int32_t col_unique_id, IndexType index_type, std::shared_ptr<TabletIndex>& res) const;
```

Comment:
I think return TabletIndex* is enough. And client will make sure that they can only use the return TabletIndex within the duration of this TabletSchema.